### PR TITLE
Update `app-configuration` module to store connection strings in Key Vault

### PR DESCRIPTION
**Pull request by MikeLarah**
Also:
- Fixes issue in `key-vault` module where deployment would fail if `enableDiagnostics` was set to `false`
- Fixes incorrect parameter name in `storage-account`

### CODE REVIEW

File: modules/general/app-configuration/main.bicep [hunk 1]
```diff
@@ -45,6 +63,26 @@ resource app_config_store 'Microsoft.AppConfiguration/configurationStores@2020-0
   tags: resourceTags
 }
 
+module connection_string_secret '../key-vault-secret/main.bicep' = if (saveConnectionStringsToKeyVault) {
+  name: 'appConfigConnStrSecretDeploy${name}'
+  scope: resourceGroup(keyVaultSubscriptionId, keyVaultResourceGroupName) 
+  params: {
+    keyVaultName: keyVaultName
+    secretName: keyVaultConnectionStringSecretName
+    contentValue: listKeys(name, '2020-06-01').value[0].connectionString
```
Comment (JamesDawson) [inline modules/general/app-configuration/main.bicep:72]: Will `name` resolve properly if the Key Vault is in a different resource group to the AppConfig store?

Maybe we can reference the AppConfig store directly instead? (also avoids having to keep the API versions in-sync)

```
var acsKey = useExisting ? existing_app_config_store.listKeys().value[0] : app_config_store.listKeys().value[0]
module ... {
  params: {
    contentValue: acsKey.connectionString
  }
}
```

File: modules/general/app-configuration/main.bicep [hunk 2]
```diff
@@ -45,6 +63,26 @@ resource app_config_store 'Microsoft.AppConfiguration/configurationStores@2020-0
...
+  }
+}
+
+module readonly_connection_string_secret '../key-vault-secret/main.bicep' = if (saveConnectionStringsToKeyVault) {
+  name: 'appConfigRoConnStrSecretDeploy${name}'
+  scope: resourceGroup(keyVaultSubscriptionId, keyVaultResourceGroupName) 
+  params: {
+    keyVaultName: keyVaultName
+    secretName: keyVaultReadOnlyConnectionStringSecretName
+    contentValue: listKeys(name, '2020-06-01').value[2].connectionString
```
Comment (JamesDawson) [inline modules/general/app-configuration/main.bicep:82]: as above

File: modules/general/key-vault/diagnostics.bicep [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+
+resource key_vault 'Microsoft.KeyVault/vaults@2021-06-01-preview' existing = {
+  name: keyVaultName
+}
+
+var _diagnosticsStorageAccountName = empty(diagnosticsStorageAccountName) ? keyVaultName : diagnosticsStorageAccountName
+
+
+resource existing_storage_account 'Microsoft.Storage/storageAccounts@2021-06-01' existing = if (useExistingStorageAccount) {
```
Comment (JamesDawson) [inline modules/general/key-vault/diagnostics.bicep:23]: It's probably best as a separate issue/change, but for completeness do you think we should support using an existing storage account in a different resource group?

Comment (MikeLarah) [reply modules/general/key-vault/diagnostics.bicep:23]: I've added a new issue https://github.com/endjin/Endjin.RecommendedPractices.Bicep/issues/45

**Review by JamesDawson — APPROVED** [review]

**Pull request merged**